File: app/src/main/java/nl/acidcats/tumblrlikes/ui/screens/load_likes_screen/LoadLikesScreenContract.java
package nl.acidcats.tumblrlikes.ui.screens.load_likes_screen;

import nl.acidcats.tumblrlikes.ui.screens.base.BasePresenter;
import nl.acidcats.tumblrlikes.ui.screens.base.BaseView;

/**
 * Created on 18/10/2018.
 */
public interface LoadLikesScreenContract {

    interface Presenter extends BasePresenter<LoadLikesScreenContract.View> {
        void onViewCreated();

        void cancelLoading();

        void skipLoading();

        void retryLoading();

        void showSettings();
    }

    interface View extends BaseView {
        void showLoadProgress(int pageCount, long totalPhotoCount);

        void showErrorAlert(int errorStringId);

        void showAllLikesLoaded(long count);

        void showLoadingCancelled();
    }
}

File: app/src/main/java/nl/acidcats/tumblrlikes/ui/screens/load_likes_screen/LoadLikesScreenContract.kt
package nl.acidcats.tumblrlikes.ui.screens.load_likes_screen

import nl.acidcats.tumblrlikes.core.constants.LoadLikesMode
import nl.acidcats.tumblrlikes.ui.screens.base.BasePresenter
import nl.acidcats.tumblrlikes.ui.screens.base.BaseView

/**
 * Created on 18/10/2018.
 */
interface LoadLikesScreenContract {

    object Keys {
        const val KEY_MODE = "mode"
    }

    interface Presenter : BasePresenter<LoadLikesScreenContract.View> {
        fun onViewCreated(mode:LoadLikesMode = LoadLikesMode.SINCE_LAST)

        fun cancelLoading()

        fun skipLoading()

        fun retryLoading()

        fun showSettings()
    }

    interface View : BaseView {
        fun showLoadProgress(pageCount: Int, totalPhotoCount: Int)

        fun showErrorAlert(errorStringId: Int)

        fun showAllLikesLoaded(count: Int)

        fun showLoadingCancelled()
    }
}
